### PR TITLE
[PUB-2795] Add Org Switcher to Publish UI

### PR DESCRIPTION
**Pull request by hamstu**
## Description

https://buffer.atlassian.net/browse/PUB-2795

## Context & Notes

- [x] Only show switcher for users with feature flip
- [x] Only show when num. orgs >= 2
- [x] Bug: Reloading after switching orgs does not work right (defaults to first org.)

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/organizations/middleware.js [hunk 1]
```diff
@@ -32,39 +36,56 @@ export default ({ dispatch, getState }) => next => action => {
...
+        p => p.organizationId === action.id
+      );
+      const firstProfileInOrganization = profilesInOrganization[0];
+      const newPath = profilePages.getRoute({
+        profileId: firstProfileInOrganization.id,
+      });
+      dispatch(push(newPath));
+      dispatch(
+        profileActions.selectProfile({ profile: firstProfileInOrganization })
+      );
```
Comment (hamstu) [inline packages/organizations/middleware.js:84]: I don't love this logic here. Can't help but wonder if there's a better way to do it, but I think that all comes back to a rewrite/overhaul in our routing. (I.e., Ideally we could just do the push/navigate and everything else would work, but without the `selectProfile` it doesn't.)

Comment (hamstu) [reply packages/organizations/middleware.js:84]: It's also worth noting that things don't **not** work on page reload yet. For example, if you switch to a new org/profile and reload the URL remains correct but the profile queue doesn't ever load. 

Will add to the to dos on this PR.

Comment (anafilipadealmeida) [reply packages/organizations/middleware.js:84]: Hey @hamstu, I'm curious, why isn't your org switcher itemClick `switchOrganization` not just `organization.goTo({orgId})` from routes ? In theory that should be enough to trigger everything else in the app.

File: packages/app-shell/index.js [hunk 2]
```diff
@@ -2,9 +2,13 @@ import { connect } from 'react-redux';
 import {
   newBusinessTrialists,
   preferencesGeneral,
+  organization,
 } from '@bufferapp/publish-routes';
 import { actions as modalActions } from '@bufferapp/publish-modals';
 import { getURL } from '@bufferapp/publish-server/formatters/src';
+
+// import { actions as orgActions } from '@bufferapp/publish-data-organizations';
```
Comment (anafilipadealmeida) [inline packages/app-shell/index.js:10]: Would we be able to delete this? 🙂

File: packages/organizations/middleware.js [hunk 3]
```diff
@@ -2,9 +2,11 @@ import {
   actions as dataFetchActions,
   actionTypes as dataFetchActionTypes,
 } from '@bufferapp/async-data-fetch';
+import { organization } from '@bufferapp/publish-routes';
```
Comment (anafilipadealmeida) [inline packages/organizations/middleware.js:5]: I believe we might not need this one anymore

**Pull request merged**